src/prisma-service.ts

```diff
@@ -0,0 +1,956 @@
+            baseQuery,
+            findMany,
+            transformResponse
+          ),
+        (reason) => reason
+      )
+    ),
+    TE.match(
+      (reason) => handleError(reason),
+      (response) => response
+    )
+  );
+  return await task();
+};
+
+const safeNumber = (num, defaultValue) => {
+  return typeof num === 'number' && !isNaN(num) ? num : defaultValue;
+};
+
+const buildPageQuery = <T>(pagination: Pagination<T>, nextToCursor) => {
+  const pageSize = safeNumber(pagination.size, maxSizeLimit);
+  const incrementedSize = pageSize + 1;
+
+  if (
+    pagination &&
+    'searchSince' in pagination &&
+    pagination.searchSince &&
+    'searchDirection' in pagination &&
+    pagination.searchDirection
+  ) {
+    const page =
+      pagination.searchDirection === SearchDirection.Before
+        ? -incrementedSize
+        : incrementedSize;
+    return {
+      take: page,
+      cursor: pagination.searchSince
+    };
+  } else if (pagination && 'next' in pagination) {
+    return { take: incrementedSize, cursor: nextToCursor(pagination.next) };
+  }
+  if (pagination && pagination.size) {
+    return { take: incrementedSize };
+  } else return { take: incrementedSize };
+};
+
+export const paginatedQuery = async (
+  pagination,
+  nextToCursor,
+  cursorToNext,
+  baseQuery,
+  findMany,
+  transformResponse
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const pageQueryParams = buildPageQuery(pagination, nextToCursor);
+
+    const pagedQuery = {
+      ...baseQuery,
+      ...(pagination ? pageQueryParams : {})
+    };
+
+    const rawResults = await findMany(pagedQuery);
+
+    const pageSize = pageQueryParams.take
+      ? pageQueryParams.take - 1
+      : maxSizeLimit;
+    const hasNextPage = rawResults.length > pageSize;
+
+    let result = rawResults;
+    let next;
+    if (hasNextPage) {
+      result = rawResults.slice(0, -1);
+      const last = rawResults.at(-1);
+      next = cursorToNext(last);
+    }
+    return respond(result, transformResponse, next);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+const toCreatedAtCursor = (row) => ({ created_at: new Date(row.created_at) });
+const fromCreatedAtCursor = (row) => ({
+  created_at: row.created_at.toISOString()
+});
+
+const toOrdinalCursor = (row) => ({ ordinal: BigInt('0x' + row.ordinal) });
+const fromOrdinalCursor = (row) => ({ ordinal: row.ordinal.toString(16) });
+
+const toCreatedAtOrdinalCursor = (row) => ({
+  ...toCreatedAtCursor(row),
+  ...toOrdinalCursor(row)
+});
+const fromCreatedAtOrdinalCursor = (row) => ({
+  ...fromCreatedAtCursor(row),
+  ...fromOrdinalCursor(row)
+});
+
+export const globalSnapshots = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return await paginatedResults(
+    event,
+    toCreatedAtOrdinalCursor,
+    fromCreatedAtOrdinalCursor,
+    {
+      include: { dag_blocks: true },
+      orderBy: [
+        { created_at: Prisma.SortOrder.desc },
```

Comment:
Why do we need to sort by both `created_at` and `ordinal` here? Shouldn't `ordinal` be enough and be more efficient?